### PR TITLE
improve PlaintextPasswordEncoder docBlock summary

### PR DESCRIPTION
**Pull request by jrushlow**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | 

Updates class summary as suggested in tkt #35927 & pr #35929 to suggest the encoder is for test usage.

### CODE REVIEW

**Comment by jrushlow** [comment]
Oops, had this ready yesterday but forgot to hit the submit button. I choose the master branch as I didn't think this warranted a merge in earlier branches. If that would be more appropriate, let me know and I'll change it..

Thanks for the feed back in the previous PR & issue

File: src/Symfony/Component/Security/Core/Encoder/PlaintextPasswordEncoder.php [hunk 1]
```diff
@@ -14,7 +14,7 @@
 use Symfony\Component\Security\Core\Exception\BadCredentialsException;
 
 /**
- * PlaintextPasswordEncoder does not do any encoding.
+ * PlaintextPasswordEncoder does not do any encoding but is useful in testing environments.
  *
```
Comment (jrushlow) [inline src/Symfony/Component/Security/Core/Encoder/PlaintextPasswordEncoder.php]: "As this encoder is not cryptographically secure, usage of it in production environments is not recommended"

Thoughts as an added description?

**Comment by dmaicher** [comment]
> Oops, had this ready yesterday but forgot to hit the submit button. I choose the master branch as I didn't think this warranted a merge in earlier branches. If that would be more appropriate, let me know and I'll change it..
> 
> Thanks for the feed back in the previous PR & issue

@jrushlow the PR should target the lowest maintained symfony branch where this should be applied :wink: So would be good to rebase it and change the target to 3.4

https://github.com/symfony/symfony/blob/3.4/src/Symfony/Component/Security/Core/Encoder/PlaintextPasswordEncoder.php#L17

The maintainers will then merge this fix into the upper version branches

**Review by dmaicher — APPROVED** [review]

File: src/Symfony/Component/Security/Core/Encoder/PlaintextPasswordEncoder.php (hunk 1 above)
Comment (xabbuh) [reply src/Symfony/Component/Security/Core/Encoder/PlaintextPasswordEncoder.php]: I would go one steps further and reword it to "[...] usage of it in production environments is discouraged."

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @jrushlow.
